aquarius/events/processors.py

```diff
@@ -139,14 +131,50 @@ def get_tokens_info(self, record):
             datatokens.append(
                 {
                     "address": service["datatokenAddress"],
-                    "name": token_contract.caller.name(),
-                    "symbol": token_contract.caller.symbol(),
+                    "name": self.get_token_name(token_contract),
+                    "symbol": self.get_token_symbol(token_contract),
                     "serviceId": service["id"],
                 }
             )
 
         return datatokens
 
+    def get_token_name(self, contract):
```

Comment:
ok, I get it, but we can make these a single function with getattr :) Otherwise we are not being DRY at all, and crowding processors.

Reply:
agreed, I will improve it